### PR TITLE
Add ability to disable state storage locally or globally

### LINKED ISSUE #1812: Allow for disabling of `[jobs]` filter form state saving functionality
This will allow site creators to disable the state saving functionality, until, for example, cookies are explicitly allowed by the visitor.

We could add a check for a `disable-local-storage` or `disable-cookies` class or something similar on the body tag?  There might be a best practice for this as well.

### PR DESCRIPTION
**Pull request by jom**
Fixes #1812 

#### Changes proposed in this Pull Request:

* Cleanup: Remove method prefixes. Not necessary in scope.
* Adds new filter so site admins or plugins can add data attributes to wrapper for `[jobs]` shortcode.
* Adds ability to disable the form/results state storage on `[jobs]` both locally with data attribute or globally with class on `<body>`. This could be useful for cookie consent customization or if the theme provides an alternative approach.

#### Testing instructions:

Test both of these snippets individually and make sure when activated the `[jobs]` filter form doesn't store anything in local session and when you click onto a job listing result and then click back nothing is preserved (some browsers might automatically preserve the form state).

* Test with this snippet and make sure state isn't saved:
```php
add_filter( 'job_manager_jobs_shortcode_data_attributes', function( $data ) {
  $data['disable-form-state-storage'] = '1';
  
  return $data;
} );
```

* Test with this snippet and make sure state isn't saved:
```php
add_filter( 'body_class', function( $classes ) {
  $classes[] = 'disable-job-manager-form-state-storage';
  
  return $classes;
} );
```
* Test with both snippets removed. Click on job listing in `[jobs]` results and then click back. Make sure form and results are preserved and the results aren't reloaded.

### CODE REVIEW

File: assets/js/ajax-filters.js [hunk 1]
```diff
@@ -1,15 +1,39 @@
...
+			return false;
+		}
+
+		// Check to see if it is globally disabled.
+		if ( $( document.body ).hasClass( 'disable-job-manager-form-state-storage' ) ) {
+			return false;
+		}
+
+		// Check if it is disabled on this specific element.
+		if ( $target.data( 'disable_form_state_storage' ) ) {
```
[the remark this answers is not in the document]
Comment (jom) [reply assets/js/ajax-filters.js]: Done!

[the remark this answers is not in the document]
Comment (jom) [reply assets/js/ajax-filters.js]: Done in https://github.com/Automattic/WP-Job-Manager/pull/1813/commits/85ae795dd3c811a11b7dea9cb6ae74f713fbbcd6 and update snippets in testing instructions.

**Comment by jom** [comment]
@alexsanford I think we can add the snippets from testing steps to the documentation if it comes up. This functionality is hopefully less controversial now that is really just about restoring state when coming back after clicking on a job listing in the results.

**Comment by MarieComet** [comment]
Hi @jom, when that will be released on wordpress.org ?
Thanks.

**Pull request merged**